Core/Registry/RegistryManager.cs

```diff
@@ -49,6 +60,40 @@ private RegistryManager(string path, KSP ksp)
             }
         }
 
+        public void Dispose()
+        {
+            // Release the lock.
+            ReleaseLock();
+        }
+
+        /// <summary>
+        /// Tries to lock the registry by creating a lock file.
+        /// </summary>
+        /// <returns><c>true</c>, if lock was gotten, <c>false</c> otherwise.</returns>
+        public bool GetLock()
+        {
+            if (File.Exists(lock_file_path))
```

Comment:
Oooh! This looks like a race condition to me!

* Process A checks to see if the lockfile exists, and discovers it doesn't.
* Process B creates the lockfile
* Process A calls `File.Create`, but it's already there!  Bam!

Further discussion regarding locking incoming on the main PR thread. :)